bin/Audio_Transcript.py

```diff
@@ -0,0 +1,111 @@
+import sys
+import os
+import glob
+import webrtcvad
+import logging
+import waveSplit
+from deepspeech.model import Model
+import scipy.io.wavfile as wav
+from timeit import default_timer as timer
+
+
+# Debug helpers
+logging.basicConfig(stream=sys.stderr, level=logging.DEBUG)
+
+
+def main(args):
+    if len(args) != 2:
+        sys.stderr.write(
+            'Usage: example.py <aggressiveness> <path to wav files>\n')
+        sys.exit(1)
+
+    N_FEATURES = 26
+    N_CONTEXT = 9
+    BEAM_WIDTH = 500
+    LM_WEIGHT = 1.75
+    VALID_WORD_COUNT_WEIGHT = 1.00
+    WORD_COUNT_WEIGHT = 1.00
+
+    # Run deepspeech on the directory containing wav files
+    dirName = input("Enter the deepspeech models directory path: ")
+    dirName = os.path.expanduser(dirName)
+
+    models = glob.glob(dirName + "/*.pb")[0]
+    logging.debug("Found Model: %s" % models)
+
+    alphabet = glob.glob(dirName + "/alphabet.txt")[0]
+    logging.debug("Found Alphabet: %s" % alphabet)
+
+    lm = glob.glob(dirName + "/lm.binary")[0]
+    trie = glob.glob(dirName + "/trie")[0]
+    logging.debug("Found Language Model: %s" % lm)
+    logging.debug("Found Trie: %s" % trie)
+
+    title_names = ['Filename', 'Duration(s)', 'Inference Time(s)', 'Model Load Time(s)', 'LM Load Time(s)']
+    print("\n%-30s %-20s %-20s %-20s %s" % (title_names[0], title_names[1], title_names[2], title_names[3], title_names[4]))
+
+    wav_files = glob.glob(args[1] + "/*.wav")
+    for single_wav_file in wav_files:
+        logging.debug("Processing: %s" % single_wav_file)
+        audio, sample_rate = waveSplit.read_wave(single_wav_file)
+        vad = webrtcvad.Vad(int(args[0]))
+        frames = waveSplit.frame_generator(30, audio, sample_rate)
+        frames = list(frames)
+        segments = waveSplit.vad_collector(sample_rate, 30, 300, vad, frames)
+
+        model_time = 0.0
+        lm_time = 0.0
+        inference_time = 0.0
+        audio_time = 0.0
+
+        f = open(single_wav_file.rstrip(".wav") + ".txt", 'w')
+        logging.debug("Transcript saved @ %s" % (single_wav_file.rstrip(".wav") + ".txt"))
+
+        for i, segment in enumerate(segments):
+            # Run VAD on the input file
+            audio = 'chunk-%002d.wav' % (i,)
```

Comment:
That's going to fill the disk and generate useless I/O. You can probably rely on a byte buffer? I guess it's a left over from your previous refactoring, passing files made sense since you were calling the binaries :)

Reply:
Nice Idea :) 
Will do that...